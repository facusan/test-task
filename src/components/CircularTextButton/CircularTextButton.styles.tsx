import styled from "styled-components";

export const CircularTextButtonWrapper = styled.div`
  padding: 15px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  &:hover {
    cursor: pointer;
  }
`;
export const Circle = styled.div`
  border-radius: 25px;
  height: 50px;
  width: 50px;
  background-color: #167eff;
  color: white;
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: x-large;
  margin: 10px;
`;
export const TextWrapper = styled.div`
  font-family: sans-serif;
  font-weight: bold;
`;
