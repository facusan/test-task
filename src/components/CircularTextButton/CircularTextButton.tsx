import { MouseEventHandler } from "react";
import {
  Circle,
  CircularTextButtonWrapper,
  TextWrapper,
} from "./CircularTextButton.styles";

interface CircularTextButtonProps {
  icon: JSX.Element;
  text: string;
  onClick: MouseEventHandler<HTMLDivElement>;
}
export default function CircularTextButton({
  text,
  icon,
  onClick,
}: CircularTextButtonProps) {
  return (
    <CircularTextButtonWrapper onClick={onClick}>
      <Circle>{icon}</Circle>
      <TextWrapper>{text}</TextWrapper>
    </CircularTextButtonWrapper>
  );
}
