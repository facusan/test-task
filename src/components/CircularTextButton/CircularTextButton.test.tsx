import renderer from "react-test-renderer";
import CircularTextButton from "./CircularTextButton";
import { HiArrowNarrowUp } from "react-icons/hi";

it("renders CircularTextButton correctly", () => {
  const component = renderer.create(
    <CircularTextButton
      onClick={() => {}}
      icon={<HiArrowNarrowUp />}
      text="Send"
    ></CircularTextButton>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
