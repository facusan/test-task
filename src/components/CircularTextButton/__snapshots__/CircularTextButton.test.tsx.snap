// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CircularTextButton correctly 1`] = `
<div
  className="sc-bczRLJ dRwWsE"
  onClick={[Function]}
>
  <div
    className="sc-gsnTZi klUVPW"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 20 20"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M5.293 7.707a1 1 0 010-1.414l4-4a1 1 0 011.414 0l4 4a1 1 0 01-1.414 1.414L11 5.414V17a1 1 0 11-2 0V5.414L6.707 7.707a1 1 0 01-1.414 0z"
        fillRule="evenodd"
      />
    </svg>
  </div>
  <div
    className="sc-dkzDqf hSBLhr"
  >
    Send
  </div>
</div>
`;
